src/rewards/poprewards.cpp

```diff
@@ -0,0 +1,115 @@
+#include <cassert>
+#include <vector>
+#include <veriblock/entities/atv.hpp>
+#include <veriblock/rewards/poprewards.hpp>
+#include <veriblock/rewards/poprewards_calculator.hpp>
+
+namespace altintegration {
+
+static int getBestPublicationHeight(
+    const std::vector<VbkEndorsement>& endorsements,
+    const VbkBlockTree& vbk_tree) {
+  int bestPublication = -1;
+  for (const VbkEndorsement& e : endorsements) {
+    auto* block = vbk_tree.getBlockIndex(e.blockOfProof);
+    if (block == nullptr) continue;
```

Comment:
do you need to check that block **exists** or **exists on best chain**?

Reply:
best chain, I guess

Reply:
then, you need to do 
```
auto* block = vbk_tree.getBlockIndex(e.blockOfProof);
if(vbk_tree.getBestChain().contains(block)) {
   // block exists and is on a main chain
}
```